pkgs/os-specific/linux/plymouth/default.nix

```diff
@@ -1,61 +1,94 @@
-{ stdenv, fetchurl, autoreconfHook, pkgconfig, libxslt, docbook_xsl
+{ stdenv, fetchFromGitLab
+, pkgconfig, autoreconfHook, libxslt, docbook_xsl, intltool
 , gtk3, udev, systemd
+, withLabelFt ? true, freetype ? null
 }:
 
+assert withLabelFt -> freetype != null;
+
 stdenv.mkDerivation rec {
   pname = "plymouth";
-  version = "0.9.4";
+  version = "2020-04-07";
 
-  src = fetchurl {
-    url = "https://www.freedesktop.org/software/plymouth/releases/${pname}-${version}.tar.xz";
-    sha256 = "0l8kg7b2vfxgz9gnrn0v2w4jvysj2cirp0nxads5sy05397pl6aa";
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "plymouth";
+    repo = "plymouth";
+    rev = "6ca4b5b24d02242e85a3a5a4727aa195d4008210";
+    sha256 = "1l25wqr6wwbiy7ivnl766prj1ndfc9xdprw9ya7i67yhrdi429cz";
   };
 
   nativeBuildInputs = [
-    autoreconfHook pkgconfig libxslt docbook_xsl
+    pkgconfig autoreconfHook
+    libxslt docbook_xsl intltool
   ];
 
   buildInputs = [
-    gtk3 udev systemd
+    gtk3 udev systemd freetype
+  ];
+
+  patches = [
+    # https://build.opensuse.org/package/show/openSUSE%3AFactory/plymouth
+    ./manpages.patch
+    # ./only_use_fb_for_cirrus_bochs.patch
+  ] ++ stdenv.lib.optionals withLabelFt [
+    # https://gitlab.freedesktop.org/plymouth/plymouth/-/issues/45
+    ./v3-0001-ply-label-Don-t-crash-if-label-plugin-fails.patch
```

Comment:
Is this patch set necessary for flicker-free experience? I would rather not include unmerged patches using legacy libraries (freetype).

Reply:
It isn't necessary but without it, some of the functionality for some themes will be impacted, anything that uses `display-message` won't be shown since we don't bundle the original `label.so` since it's far more bloated than `label-ft.so`.

But if it's just the password prompt (basically a pile of images) and the background, then no it isn't needed.